packages/rest/src/spec-enhancers/consolidate.spec-enhancer.ts

```diff
@@ -60,42 +64,63 @@ const debug = debugFactory('loopback:openapi:spec-enhancer:consolidate');
  */
 @injectable(asSpecEnhancer, {scope: BindingScope.SINGLETON})
 export class ConsolidationEnhancer implements OASEnhancer {
-  name = 'consolidate';
-  disabled: boolean;
+  readonly name = 'consolidate';
+
+  private _disabled?: boolean;
+
+  get disabled(): ValueOrPromise<boolean> {
+    if (this._disabled !== undefined) return this._disabled;
+
+    return (async () => {
+      const consolidate: boolean | undefined = (await this._config.values())[0]
+        ?.rest?.openApiSpec?.consolidate;
+      if (consolidate === undefined) return false;
+      this._disabled = consolidate === false;
+      return this._disabled;
+    })();
+  }
 
   constructor(
-    @inject(CoreBindings.APPLICATION_CONFIG, {optional: true})
-    readonly config?: ApplicationConfig,
+    @inject.view(filterByKey(CoreBindings.APPLICATION_CONFIG.key))
+    private _config: ContextView<ApplicationConfig>,
```

Comment:
Should move this to property injection.

Reply:
Are you modifying  the _config object afterwards?

Reply:
ContextView would allow modification of the configuration after the ConsolidationEnhancer is initialised, which would give a better developer experience instead of having to re-bind the enhancer for those who need a more "dynamic" configuration.